cmplibrary/src/main/java/com/sourcepoint/cmplibrary/data/ServiceImpl.kt

```diff
@@ -150,7 +150,7 @@ private class ServiceImpl(
                 }
                 .executeOnRight { metaDataResponse -> handleMetaDataResponse(metaDataResponse) }
 
-            if (messageReq.authId != null || campaignManager.shouldCallConsentStatus) {
+            if (messageReq.authId != null || campaignManager.shouldCallConsentStatus || campaignManager.requiresNewConsentData) {
```

Comment:
The `requiresNewConsentData` condition is part of the logic contained into `shouldCallConsentStatus`, please move requiresNewConsentData into shouldCallConsentStatus